express-post-json/server.ts

```diff
@@ -0,0 +1,37 @@
+import express from 'express';
+
+type Student = {
+  id?: number;
+  name: string;
+  course: string;
+  score: number;
+};
+
+type Students = Record<number, Student>;
+
+let nextID: number = 1;
+const grades: Students = {};
+
+const app = express();
+
+app.get('/api/grades', (req, res) => {
+  const gradesResponse: Student[] = [];
+  for (const key in grades) {
+    gradesResponse.push(grades[key]);
+  }
+  res.json(gradesResponse);
+});
+
+const jsonParser = express.json();
+app.use(jsonParser);
```

Comment:
Typically we do this at the top of the file, right after creating `app`. This is because it's easy to get put endpoints that require it before the call, and that leads to hard to find bugs. Also, the JSON middleware does nothing if the content-type is not 'application/json' so there's no harm in having it run on every request.